web/templates/recipient_row.hbs

```diff
@@ -28,7 +28,7 @@
             <a class="message_label_clickable narrows_by_topic tippy-narrow-tooltip"
               href="{{topic_url}}"
               data-tippy-content="{{t 'Go to #{display_recipient} > {topic_display_name}' }}">
-                {{#if use_match_properties}}
+                {{#if (and use_match_properties (not is_empty_string_topic))}}
                     <span class="stream-topic-inner">{{{match_topic}}}</span>
```

Comment:
Should we do it in a more explicit way i.e. processing each message of the search result and then adding one more field named `match_topic_display_name` to get used here.

```hbs
{{#if use_match_properties}}
   <span class="stream-topic-inner {{#if is_empty_string_topic}}empty-topic-display{{/if}}">{{{match_topic_display_name}}}</span>
{{else}}
```

The current approach saves that processing but might be not clear enough to understand in the first read. Adding comment is also an option.

Reply:
This seems OK to me.